### PR TITLE
Render overview on the index route

### PR DESCRIPTION
**Pull request by safinn**
Solves #13 

Renders an overview page using handlebars.
Had to add the size of the files in the cache to be able to display them.

Added a few packages:

- timeago.js to convert the publish date.
- handlebars to render the page.

screen shot 2017-12-08 at 22 43 47

### CODE REVIEW

**Comment by leo** [comment]
Amazing! Looks great.

Could you please make the function for rendering the handlebars template asynchronous? In addition, I don't see a need for including `fs-extra`, you can just use `promsify(fs.readFile)` (from the native "util" module).

File: lib/view.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-module.exports = () => {
+const readFileAsync = promisify(fs.readFile)
+
+module.exports = async () => {
   let viewContent = false
   const viewPath = path.normalize(path.join(__dirname, '/../views/index.hbs'))
 
   try {
-    viewContent = fs.readFileSync(viewPath, 'utf8')
+    viewContent = await readFileAsync(viewPath, { encoding: 'utf8' })
```
Comment (leo) [inline lib/view.js]: `'utf8'` works even with the native module 😊 no need for an object

File: lib/view.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const readFileAsync = promisify(fs.readFile)
+
+module.exports = async () => {
+  let viewContent = false
+  const viewPath = path.normalize(path.join(__dirname, '/../views/index.hbs'))
+
+  try {
+    viewContent = await readFileAsync(viewPath, 'utf8')
+  } catch (err) {
+    throw err
```
Comment (leo) [inline lib/view.js]: If you don't handle the error, you can just write:

```js
const viewContent = await readFileAsync(viewPath, 'utf8')
```

Without the try/catch block 😊

Comment (leo) [reply lib/view.js]: This also means you can remove the `let viewContent = false` at the top... 🚀

File: lib/routes.js [hunk 3]
```diff
@@ -175,5 +177,30 @@ module.exports = ({ cache, config }) => {
...
   }
 
+  exports.overview = async (req, res) => {
+    try {
+      const render = await prepareView()
+
+      const details = {
+        account: config.account,
+        repository: config.repository,
+        date: timeago().format(cache.latest.pub_date),
```
Comment (leo) [inline lib/routes.js]: Let's use `date-fns` here please 😊 It's faster, slimmer and currently the best option out there.

File: views/index.hbs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      }
+
+      #version {
+        margin-right: 8px;
+        padding: 2px 6px;
+        border-radius: 4px;
+        background: linear-gradient(#ff00a5, #f65538)
+      }
+
+      #releaseNotes {
```
Comment (leo) [inline views/index.hbs]: No camelcase in CSS properties please, always dashes (also adjust the HTML) 😊

**Comment by leo** [comment]
Thank you for this wonderful PR!

**Comment by safinn** [comment]
No worries. Learnt about date-fns. Looks really good! 👍

**Pull request merged**